### PR TITLE
make sure email addresses used with 'tell' and 'killperson' exist in keyring

### PR DESCRIPTION
**Pull request by joshrabinowitz**
This PR a fix for #176. 

#176 is a bug with `tell` which adds all keys with a substring matching the specified username. For example, `git secret tell bob` will add all keys with the string `bob` anywhere in the username or email address. 

Here's the details:
 * each gpg key is associated with an `email address`, and optionally a `name`. 
 * gpg does substring matching against both the `email address` and the `username` when looking for keys matching a string.

Also, 
 * the `name`associated with a key is not guaranteed to be unique and 
 * the `email` can be a substring of other names or emails

For the above reasons, this PR changes the code so you have to specify users by email address with `git secret tell`. (Edit: The docs always said to use email addresses so there is no doc change.)

@sobolevn @simbo1905 thoughts/suggestions/input?

### CODE REVIEW

**Comment by joshrabinowitz** [comment]
This PR does still have a bug where if 
* one email address is a substring of another (for example bob@github.com and joebob@example.com), and 
* you use `git secret tell bob@github.com`, 
then it will 
* add keys for  both users and not just the one you probably meant. 

I don't see an easy way to fix that and figured we could tackle that in a separate PR. If you see the right way to fix this issue let me know @sobolevn @simbo1905  @dave-pollock

File: src/commands/git_secret_tell.sh [hunk 1]
```diff
@@ -64,6 +64,21 @@ function tell {
     _abort "you must provide at least one email address."
   fi
 
+  # TODO: this block is duplicated in git_secret_killperson.sh and should be factored
```
Comment (sobolevn) [inline src/commands/git_secret_tell.sh]: Should we refactor it now or you feel like it's not the time yet?

File: src/commands/git_secret_killperson.sh [hunk 2]
```diff
@@ -24,10 +24,25 @@ function killperson {
...
+  local gpg_uids
+  gpg_uids=$(_get_users_in_gpg_keyring "$secrets_dir_keys")
+  for email in "${emails[@]}"; do
+    local email_ok=0
+    for uid in $gpg_uids; do
+        if [[ "$uid" == "$email" ]]; then
+            email_ok=1
+        fi
+    done
+    if [[ "$email_ok" == 0 ]]; then
```
[the remark this answers is not in the document]
Comment (joshrabinowitz) [reply src/commands/git_secret_killperson.sh]: will revisit, thanks @simbo1905

File: src/commands/git_secret_tell.sh (hunk 1 above)
Comment (joshrabinowitz) [reply src/commands/git_secret_tell.sh]: I'll take another look and let you know @sobolevn

File: src/commands/git_secret_killperson.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshrabinowitz) [reply src/commands/git_secret_killperson.sh]: fixed!

File: src/commands/git_secret_tell.sh (hunk 1 above)
Comment (joshrabinowitz) [reply src/commands/git_secret_tell.sh]: factored.

**Pull request merged**